kong/runloop/handler.lua

```diff
@@ -1284,38 +1340,23 @@ return {
       end
 
       -- clear hop-by-hop request headers:
-      local connection = var.http_connection
-      if connection then
-        local header_names = re_split(connection .. ",", [[\s*,\s*]], "djo")
-        if header_names then
-          for i=1, #header_names do
-            if header_names[i] ~= "" then
-              local header_name = lower(header_names[i])
-              -- some of these are already handled by the proxy module,
-              -- proxy-authorization being an exception that is handled
-              -- below with special semantics.
-              if header_name ~= "close" and
-                 header_name ~= "upgrade" and
-                 header_name ~= "keep-alive" and
-                 header_name ~= "proxy-authorization" then
-                clear_header(header_names[i])
-              end
-            end
-          end
+      for _, header_name in csv(var.http_connection) do
+        -- some of these are already handled by the proxy module,
+        -- proxy-authorization being an exception that is handled
+        -- below with special semantics.
+        if header_name ~= "close" and
+           header_name ~= "upgrade" and
+           header_name ~= "keep-alive" and
+           header_name ~= "proxy-authorization" then
```

Comment:
Style: we usually write long `if` conditions like this:

```
if header_name ~= "close" and
   header_name ~= "upgrade" and
   header_name ~= "keep-alive" and
   header_name ~= "proxy-authorization"
then
  -- blah
end
```

to make it more readable.